### PR TITLE
[e2e-test] msisdn-gateway

### PR DESCRIPTION
**Pull request by rpappalax**
adding koki's e2e test.  will need some tweakage to accomodate country coverage, etc. but currently works as-is in qa jenkins.  @pdehaan?

### CODE REVIEW

File: msisdn-gateway/e2e-test/.gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+*~
+.DS_Store
+*.egg-info/
+*.pyc
+__pycache__/
+*.json
```
Comment (pdehaan) [inline msisdn-gateway/e2e-test/.gitignore]: This is fine here, but not sure if we want to promote these up into the global .gitignore (except that sneaky *.json one).

File: msisdn-gateway/e2e-test/LICENSE [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
```
Comment (pdehaan) [inline msisdn-gateway/e2e-test/LICENSE]: This is fine too, but kind of scary to think that different folders can be under different licenses. Not sure if we want to create some root LICENSE file where we release everything under **MPL-2.0**.

File: msisdn-gateway/e2e-test/run.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+echo "-----------------------------------Setup Complete----------------------------------"
+echo "---------------------------Virtual Environment Activated---------------------------"
+python ./test/control-script.py -a ${ACCOUNT_AUTH} -n ${NUMBERS}
+echo "------------------------------------Completed Test---------------------------------"
+echo "------------------------------------Begin Teardown---------------------------------"
+fab teardown || exit 1
+echo "-----------------------------------Finished Teardown-------------------------------"
+echo "---------------------------Virtual Environment Deactivated-------------------------"
+deactivate
+rm -fr venv#!/bin/sh
```
Comment (pdehaan) [inline msisdn-gateway/e2e-test/run.sh]: Should this line have the _shebang_ at the end?

File: msisdn-gateway/e2e-test/test/control-script.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import pexpect
+import time
+import sys
+import re
+import argparse
+import json
+from twilio.rest import TwilioRestClient
+
+host = "https://msisdn.services.mozilla.com"
+host_number = "+12182967993"
```
Comment (pdehaan) [inline msisdn-gateway/e2e-test/test/control-script.py:13]: I assume this is a legit number. Are we cool with adding it to source control here, or do we want to do some env variable thing here?

Comment (rpappalax) [inline msisdn-gateway/e2e-test/test/control-script.py:13]: koki parameterized the other nums, but you're right. we'll do same with this one

**Comment by pdehaan** [comment]
@rpappalax @kreamkorokke This looks great!
I had 4 nits above, but none of them are serious.

@rpappalax Feel free to merge when ready and we can address nits separately in a future PR (just let me know if you want me to file separate bugs).

**Comment by rpappalax** [comment]
i'll knock off 3 of them, then let koki fix parameterizing the number.  thx @pdehaan!

**Comment by pdehaan** [comment]
@rpappalax:

File: msisdn-gateway/e2e-test/test/control-script.py (hunk 4 above)
Comment (pdehaan) [inline msisdn-gateway/e2e-test/test/control-script.py:13]: Fixed in #10

**Pull request merged**